substrate/frame/staking/src/ledger.rs

```diff
@@ -106,18 +106,36 @@ impl<T: Config> StakingLedger<T> {
 	/// This getter can be called with either a controller or stash account, provided that the
 	/// account is properly wrapped in the respective [`StakingAccount`] variant. This is meant to
 	/// abstract the concept of controller/stash accounts from the caller.
+	///
+	/// Returns [`Error::BadState`] when a bond is in "bad state". A bond is in a bad state when a
+	/// stash has a controller which is bonding a ledger associated with another stash.
 	pub(crate) fn get(account: StakingAccount<T::AccountId>) -> Result<StakingLedger<T>, Error<T>> {
```

Comment:
One issue I feel when reading this function, since it handles both reading by controller and stake, it becomes a little harder to reason about which checks we need for each. What do you think about just splitting these flows.

So internally two functions `get_by_stash` and `get_by_controller` that this function calls.

Leave that decision to you though. Feel free to merge if you are happy and we probably will do lot more cleanups and refactors once controllers are removed completely.

Reply:
> probably will do lot more cleanups and refactors once controllers are removed completely.

Yes, I agree with this statement. The idea of this fn was to keep the code refactors to a minimal once the controllers were deprecated and remove the whole `StakingAccount` machinery etc.